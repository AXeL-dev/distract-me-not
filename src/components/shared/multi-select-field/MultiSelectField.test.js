import { render, screen, fireEvent } from "@testing-library/react";
import { MultiSelectField } from ".";

const options = [
  { label: 'Option 1', value: 'option-1' },
  { label: 'Option 2', value: 'option-2' },
  { label: 'Option 3', value: 'option-3' },
];

it('renders correctly', () => {
  const { asFragment } = render(<MultiSelectField label="My field" options={options} />);
  expect(asFragment()).toMatchSnapshot();
});

it('handles selection change', () => {
  const handleChange = jest.fn();
  render(<MultiSelectField label="My field" options={options} onChange={handleChange} />);
  const selectMenu = screen.getByRole('button');
  fireEvent.click(selectMenu);
  // check if all options are visible
  for (let option of options) {
    expect(screen.getByText(option.label)).toBeInTheDocument();
  }
  // select 2nd option
  fireEvent.click(screen.getByText(options[2].label));
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(selectMenu).toHaveTextContent(options[2].label);
});
