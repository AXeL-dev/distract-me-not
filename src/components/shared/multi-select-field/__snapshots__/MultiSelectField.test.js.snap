// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-nil ub-box-szg_border-box"
  >
    <div
      class="css-nil ub-dspl_flex ub-box-szg_border-box"
    >
      <div
        class="css-nil ub-dspl_flex ub-algn-itms_center ub-flx_1 ub-box-szg_border-box"
      >
        <span
          class="ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-box-szg_border-box"
        >
          My field
        </span>
      </div>
      <div
        class="css-nil ub-dspl_flex ub-algn-itms_center ub-box-szg_border-box"
      >
        <button
          aria-haspopup="true"
          class="css-69cngj ub-fnt-fam_b77syt ub-mt_0px ub-fnt-sze_12px ub-f-wght_500 ub-ln-ht_32px ub-ltr-spc_0 ub-btrr_3px ub-bbrr_3px ub-btlr_3px ub-bblr_3px ub-pt_0px ub-pb_0px ub-pr_16px ub-pl_16px ub-ml_0px ub-mr_0px ub-mb_0px ub-h_32px ub-pst_relative ub-dspl_inline-flex ub-algn-itms_center ub-flx-wrap_nowrap ub-box-szg_border-box"
          role="button"
        >
          ...
          <div
            class="ub-dspl_inline-flex ub-ml_8px ub-mr_-3px ub-box-szg_border-box"
          >
            <svg
              class="ub-w_12px ub-h_12px ub-box-szg_border-box"
              data-icon="caret-down"
              style="fill: currentColor;"
              viewBox="0 0 16 16"
            >
              <path
                d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
